smac/facade/smac_facade.py

```diff
@@ -340,16 +340,71 @@ def optimize(self):
         return incumbent
 
     def get_tae_runner(self):
+        '''
+            returns target algorithm evaluator (TAE) object
+            which can run the target algorithm given a
+            configuration
+
+            Returns
+            -------
+            smac.tae.execute_ta_run.ExecuteTARun
+        '''
         return self.solver.intensifier.tae_runner
 
     def get_runhistory(self):
+        '''
+            returns the runhistory 
+            (i.e., all evaluated configurations and the results)
+
+            Returns
+            -------
+            smac.runhistory.runhistory.RunHistory
+        '''
         if not hasattr(self, 'runhistory'):
             raise ValueError('SMAC was not fitted yet. Call optimize() prior '
                              'to accessing the runhistory.')
         return self.runhistory
 
     def get_trajectory(self):
+        '''
+            returns the trajectory 
+            (i.e., all incumbent configurations over time)
+
+            Returns
+            -------
+            List of entries with the following fields: 
+            'train_perf', 'incumbent_id', 'incumbent',
+            'ta_runs', 'ta_time_used', 'wallclock_time'
+        '''
+
         if not hasattr(self, 'trajectory'):
             raise ValueError('SMAC was not fitted yet. Call optimize() prior '
                              'to accessing the runhistory.')
         return self.trajectory
+
+    def get_X_y(self):
+        '''
+            returns all data in runhistory
+            in X, y format, 
+
+            Returns
+            ------- 
+            X: matrix of all configurations (+ instance features)
+            y vector of cost values; can include censored runs
+            cen: list of bools indicating whether the y-value is censored
+        '''
+        X = []
+        y = []
+        cen = []
+        feature_dict = self.solver.scenario.feature_dict
+        params = self.solver.scenario.cs.get_hyperparameters()
+        for k, v in self.runhistory.data.items():
+            config = self.runhistory.ids_config[k.config_id]
+            x = [config[p.name] for p in params]
```

Comment:
This isn't sorted, so you'd have to iterate over the hyperparameters of the configuration space.